eng/pipelines/sync-repos.yml

```diff
@@ -0,0 +1,21 @@
+# Refer mirror-repos.yml.
```

Comment:
We should try to update mirror-repos.yml https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/mirror-repos.yml

Reply:
The mirror-repos.ym has been updated and remove sync-repo.yml